src/Microsoft.PowerShell.Commands.Utility/commands/utility/FormatAndOutput/format-hex/Format-Hex.cs

```diff
@@ -0,0 +1,395 @@
+        [ValidateNotNullOrEmpty()]
+        public string[] Path 
+        { 
+            get { return _path; }
+            set { _path = value; }
+        }
+
+        /// <summary>
+        /// Literal path of file to process
+        /// </summary>
+        [Parameter(Mandatory = true, ParameterSetName = "LiteralPath")]
+        [ValidateNotNullOrEmpty()]
+        [Alias("PSPath")]
+        public string[] LiteralPath
+        {
+            get { return _literalPath; }
+            set { _literalPath = value; }
+        }
+
+        /// <summary>
+        /// Ojbect to process
+        /// </summary>
+        [Parameter(Mandatory = true, ParameterSetName = "ByInputObject", ValueFromPipeline = true)]
+        public PSObject InputObject
+        {
+            get { return _inputObject; }
+            set { _inputObject = value; }
+        }
+
+        /// <summary>
+        /// Type of character encoding for InputObject
+        /// </summary>
+        [Parameter(ParameterSetName = "ByInputObject")]
+        [ValidateSet("Ascii", "UTF32", "UTF7", "UTF8", "BigEndianUnicode", "Unicode")]
+        public string Encoding 
+        { 
+            get { return _encoding; }
+            set { _encoding = value; }
+        }
+
+        #endregion
+
+        #region Overrides
+
+        /// <summary>
+        /// Implements the ProcessRecord method for the FormatHex command.
+        /// </summary>
+        protected override void ProcessRecord()
+        {
+            if (String.Equals(this.ParameterSetName, "ByInputObject", StringComparison.OrdinalIgnoreCase))
+            {
+                ProcessObjectContent(_inputObject);
+            }
+            else
+            {
+                List<string> pathsToProcess = String.Equals(this.ParameterSetName, "LiteralPath", StringComparison.OrdinalIgnoreCase) ? 
+                                              ResolvePaths(_literalPath, true) : ResolvePaths(_path, false);
+                
+                ProcessPath(pathsToProcess);
+            }
+        }
+
+        #endregion
+
+        #region Paths
+
+        /// <summary>
+        /// Validate each path provided and if valid, add to array of paths to process.
+        /// If path is a literal path it is added to the array to process; we cannot validate them until we
+        /// try to process file contents.
+        /// </summary>
+        /// <param name="path"></param>
+        /// <param name="literalPath"></param>
+        /// <returns></returns>
+        private List<string> ResolvePaths(string[] path, bool literalPath)
+        {
+            List<string> pathsToProcess = new List<string>();
+            ProviderInfo provider = null;
+
+            if (literalPath)
+            {
+                foreach (string currentPath in path)
+                {
+                    string newPath = Context.SessionState.Path.GetUnresolvedProviderPathFromPSPath(currentPath);
+                    pathsToProcess.Add(newPath);
```

Comment:
do you need to validate that these paths are filesystem provider paths?
for example, `format-hex -literalpath hklm:/software` should perhaps produce the same error as `format-hex -path hklm:/software`